### PR TITLE
Dto converting feature

### PR DESCRIPTION
**Pull request by Demis05**

### CODE REVIEW

File: src/main/java/com/exadel/fedorov/orders/controllers/OrderController.java [hunk 1]
```diff
@@ -29,28 +27,30 @@
...
         }
-        return new ResponseEntity<>(orders, HttpStatus.OK);
+        return new ResponseEntity(orders, HttpStatus.OK);
     }
 
     @GetMapping(value = "/{id}")
-    public ResponseEntity<Order> read(@PathVariable("id") Long orderId) {
+    public ResponseEntity<RespOrderItemDTO> read(@PathVariable("id") Long orderId) {
         if (orderId == null) {
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```
Comment (MedinskyAlexander) [inline src/main/java/com/exadel/fedorov/orders/controllers/OrderController.java:48]: add error reason to response

File: src/main/java/com/exadel/fedorov/orders/controllers/OrderController.java [hunk 2]
```diff
@@ -29,28 +27,30 @@
...
         if (orders.isEmpty()) {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
         }
-        return new ResponseEntity<>(orders, HttpStatus.OK);
+        return new ResponseEntity(orders, HttpStatus.OK);
     }
 
     @GetMapping(value = "/{id}")
-    public ResponseEntity<Order> read(@PathVariable("id") Long orderId) {
+    public ResponseEntity<RespOrderItemDTO> read(@PathVariable("id") Long orderId) {
```
Comment (MedinskyAlexander) [inline src/main/java/com/exadel/fedorov/orders/controllers/OrderController.java]: Why does you return order item? It should be order with order item inside.

File: src/main/java/com/exadel/fedorov/orders/service/OrderService.java [hunk 3]
```diff
@@ -1,42 +1,104 @@
...
+    }
+
+    private RespOrderItemDTO convertOrderDetailsToRespItemDto(OrderDetail orderDetail) {
+        return new RespOrderItemDTO(orderDetail.getProductId(), orderDetail.getPositionPrice(), orderDetail.getProductCount());
+    }
+
+    private List<OrderDetail> convertReqOrderItemDTOsToOrderDetails(List<ReqOrderItemDTO> dtoList) {
+        return dtoList.stream()
+                .map(this::convertReqItemDtoToOrderDetail)
+//                .map(OrderItemMapper.INSTANCE::reqItemDtoToItem)
```
Comment (MedinskyAlexander) [inline src/main/java/com/exadel/fedorov/orders/service/OrderService.java]: Why don't you use map struct here?

Comment (Demis05) [reply src/main/java/com/exadel/fedorov/orders/service/OrderService.java]: Map struct does not work (spring can not find implementation of bean)
Did it with  model mapper

File: src/main/java/com/exadel/fedorov/orders/controllers/OrderController.java [hunk 4]
```diff
@@ -29,57 +26,49 @@
...
-    public ResponseEntity<Order> read(@PathVariable("id") Long orderId) {
+    public ResponseEntity<RespOrderDTO> read(@PathVariable("id") Long orderId) {
         if (orderId == null) {
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
-        Optional<Order> order = orderService.findById(orderId);
-        if (!order.isPresent()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        Optional<RespOrderDTO> orderDto = orderService.findById(orderId);
+        if (!orderDto.isPresent()) {
```
Comment (MedinskyAlexander) [inline src/main/java/com/exadel/fedorov/orders/controllers/OrderController.java]: replace it with optional.get().orElse()

**Pull request merged**